### ISSUE TITLE
lack of consistency for assimile in distros

### ISSUE THREAD

**Issue by mckaygerhard** [christopherpow/nesicide#19]
i try to package and setup a complete env for devel on gamin but this piece of software lack of any consistency standart

makefile invokes project profiles and dont let pass values, implements internal projects not directly mantained and there's no documentation, uses only qt5 (seems) libs that currently have many changes respect qt4 (more freeze)
the most depends on qscintilla, a library that in every qt change, needs recompilation.. so the only way to have a nesticle suite its a static build, and that's are not documented!

i can help but event develop newer features, need stabilize and standarize the project, so that's why a ITP are in debian as archived bug request, but remian for many years for others distros if some of this points will not change.

after several changes, many that i cannot remenber i finally compiled a static build that only depends on libc6 and phtreats, that can run in any other linux, but in some way somethings freeze and hang!..

its nonsense hav a i7 8-cores to making 8bits games, so i **proposed made ABLE this project available for olders hardware and able to build for olders UNIX environments** as qt4 and debian squeeze, _so people that have real interest_ in this king of software _can use it in their focused hardware_, and not in modern used for playstation2 hardware...  start point can be documentation, and dependencys minification (dont depends on too changing external libs)

**Comment by christopherpow** [comment]
The entire project is built within Qt Creator. Project dependencies are set up manually, true. I don't have a SUBDIRS project file, which I think would solve that.

As for QScintilla, that is included in the dependencies package which is here:
https://drive.google.com/file/d/0BwhBNo66Kd6bQjN4SlNIUUpWNDQ/view?usp=sharing
This package should be extracted into the cloned nesicide repository. The project files should already be set up to reference QScintilla and rtmidi from in there. I've only had to recompile QScintilla when I want to...not every time.

I have previously been able to get a Linux build working. But Linux is not my primary dev.

I agree there are some optimizations that would make the program better. But time is not on my side.

Bottom line, every dependency should be in that link. If not, let me know.

**Comment by mckaygerhard** [comment]
well, without prebuild binary, only guindows users will get u'r project.. its too inconsistency ...

**Comment by christopherpow** [comment]
In the past I've provided a DMG for OSX. I'm not a Linux packaging expert, but I think similar could be done there. I agree that people that insist on building everything from scratch will have no joy, but I'm not able to support everyone and everything.

**Comment by mckaygerhard** [comment]
waith a second men, i talking about  that u forced users to have custom libs only for build a only program... event if u follow propert stndars, the support of many propert os's (not only linux, i talking abou freebsd and others) will be ... but event using the detected provided by the os must be the proper way, or better not using programming libs that have a very changed api!

unless what u thinking i'm not a expert in linux, i'm a gamer, i what a linux propert nes devel ide to modify the FinalFantasy and SaintSeiya RPG games and made propers on others languaje and change some graphics to make look better.. of aybe made a more long game of -rgp

**Comment by christopherpow** [comment]
If I understand you correctly [I'm trying, I am], you're suggesting I
rewrite to avoid using libraries like QScintilla, etc.

That would be silly. The whole reason for using QScintilla is so I don't
have to recreate by myself many of the popular code editor features
provided by an open-source code editor library!

**Comment by mckaygerhard** [comment]
yeah, i review the code, qscintilla gives you that facility, but its only one of the suggested .. i'll try to packaged to deb ian to give to more people this nice project but i'ts quite difficult, due i need to customize too many the system provided libs to able..

**Comment by mckaygerhard** [comment]
i linked the RFP (procesed fro an ITP) https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=658261 if u mantain this issue open to refers what i mean..

**Comment by christopherpow** [comment]
No comments.

**Comment by mckaygerhard** [comment]
??? this issue if you normalize your build process issues like #34 due the inconsistency in detection and use standars..

**Comment by christopherpow** [comment]
I have provided AppImage deployments for Linux based on https://github.com/probonopd/linuxdeployqt. I have no time to support this beyond having done that. Can you use the AppImage?

**Comment by mckaygerhard** [comment]
that's great to a constsant changing system, but some other need to give
support to people, the problem of those alternate methods to install
packages are the poor integration to system OS so the people of such OS
like debian cannot proper do good support to thirth party desktops..

i understant that are a extra time .. but if the build process could be
more standard i inlusivelly could give help to package and integrate in
debian and alpine linux too!


Lenz McKAY Gerardo (PICCORO)
http://qgqlochekone.blogspot.com
